includes/class-wc-rest-authentication.php

```diff
@@ -66,6 +67,99 @@ protected function is_request_to_rest_api() {
 		return apply_filters( 'woocommerce_rest_is_request_to_rest_api', $woocommerce || $third_party );
 	}
 
+	/**
+	 * Check if this is a request to WCCOM Site REST API.
+	 *
+	 * @return bool
+	 */
+	protected function is_request_to_wccom_site_rest_api() {
+		if ( empty( $_SERVER['REQUEST_URI'] ) ) {
+			return false;
+		}
+
+		$rest_prefix = trailingslashit( rest_get_url_prefix() );
+		$request_uri = esc_url_raw( wp_unslash( $_SERVER['REQUEST_URI'] ) );
```

Comment:
Same as above, try and avoid accessing $_SERVER vars for building urls, use `add_query_arg( array() )` instead.

Reply:
Updated in https://github.com/woocommerce/woocommerce/pull/24123/commits/284239ee4485548486390feba1863cad51bed9cf.